src/Sulu/Bundle/AdminBundle/Resources/js/services/Translator/Translator.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+import type {TranslationMap} from './types';
+
+class Translator {
+    translations: ?TranslationMap;
+
+    set(translations: TranslationMap) {
+        this.translations = translations;
+    }
+
+    clear() {
+        this.translations = null;
+    }
+
+    translate(key: string) {
+        if (!this.translations || !(key in this.translations)) {
+            throw new Error('Translation for key "' + key + '" not found');
+        }
+
+        return this.translations[key];
+    }
+}
+
```

Comment:
I think it would make more sense to ditch the class all together if we want to do that (sounds maybe a little bit offensive, but that would be totally ok for me).

Otherwise we have build a class, which will never be used, except for one call on set. And for that we would have even more overhead on every single `translate` call, instead of a simple lookup.